controllers/component_build_controller_pac.go

```diff
@@ -181,11 +350,102 @@ func (r *ComponentBuildReconciler) UndoPaCProvisionForComponent(ctx context.Cont
 	}
 
 	// Manage merge request for Pipelines as Code configuration removal
-	mrUrl, action, err := r.UnconfigureRepositoryForPaC(ctx, component, pacSecret.Data, webhookTargetUrl)
+	baseBranch, mrUrl, action, err := r.UnconfigureRepositoryForPaC(ctx, component, pacSecret.Data, webhookTargetUrl)
 	if err != nil {
 		log.Error(err, "failed to create merge request to remove Pipelines as Code configuration from Component source repository", l.Audit, "true")
 		return "", err
 	}
+
+	// check if more components are using same repo with PaC enabled for incomings removal from repository
+	incomingsRepoTargetBranchCount := 0
+	incomingsRepoAllBranchesCount := 0
+	componentList := &appstudiov1alpha1.ComponentList{}
+	if err := r.Client.List(ctx, componentList, &client.ListOptions{Namespace: component.Namespace}); err != nil {
+		log.Error(err, "failed to list Components", l.Action, l.ActionView)
+		return "", err
+	}
+	buildStatus := &BuildStatus{}
+	for _, comp := range componentList.Items {
+		if comp.Spec.Source.GitSource.URL == component.Spec.Source.GitSource.URL {
+			buildStatus = readBuildStatus(component)
+			if buildStatus.PaC != nil && buildStatus.PaC.State == "enabled" {
+				incomingsRepoAllBranchesCount += 1
+
+				// revision can be empty and then use default branch
+				if comp.Spec.Source.GitSource.Revision == component.Spec.Source.GitSource.Revision || comp.Spec.Source.GitSource.Revision == baseBranch {
+					incomingsRepoTargetBranchCount += 1
+				}
+			}
+		}
+	}
+
+	repository, err := r.findPaCRepositoryForComponent(ctx, component)
+	if err != nil {
+		return "", err
+	}
+	incomingSecretName := ""
+	if repository != nil {
+		incomingSecretName = fmt.Sprintf("%s%s", repository.Name, pacIncomingSecretNameSuffix)
+		incomingUpdated := false
+		// update first in case there is multiple incoming entries
+		_ = updateIncoming(repository, incomingSecretName, pacIncomingSecretKey, baseBranch)
```

Comment:
Why is it safe to ignore error here? Please add a comment that explains it.

Reply:
updateIncoming isn't returning error, but bool, if it updated incomings in repository

in here I don't care, all I care about is that now there is ONLY 1 incoming